atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -406,14 +406,24 @@ enum TransactionType {
     boolean isAborted();
 
     /**
-     * Gets whether the transaction has not been committed.
+     * Gets whether the transaction has not been committed nor aborted.
      *
      * @return <code>true</code> if neither <code>commit()</code> or <code>abort()</code> have been called,
      *         otherwise <code>false</code>
      */
     @Idempotent
     boolean isUncommitted();
 
+    /**
+     * Gets whether the transaction been committed. Note that it's possible for the underlying KVS to have committed
+     * the transaction but {@link #isDefinitivelyCommitted()} to return false (e.g. when the KVS call timed out
+     * client-side but succeeded server-side).
+     *
+     * @return <code>true</code> <code>commit()</code> has been called and did not throw, otherwise <code>false</code>.
+     */
+    @Idempotent
+    boolean isDefinitivelyCommitted();
```

Comment:
I would ALMOST think that we should just expose the State enum?

ANd very clearly indicate that this is all from the perspective of "did this Transaction Java Object Instance, MANAGE to get to the point of setting it's state. So for example if the thread was interrupted at some point, as you say, the transaction may have been committed, but this will not be reflected in this object and results of these methods.

Like it might be useful to expose the enum and draw the state diagram?